### DISCUSSION TITLE
min-height 100% for scroll area

### DISCUSSION THREAD

**Discussion by BrianHung** [radix-ui/primitives#1485] [Help]
Wondering why for the ScrollArea component, min-height isn't also set to 100% like min-width is.

https://github.com/radix-ui/primitives/blob/a53790ee9d6d9be48fe84c3f72e310289c95113b/packages/react/scroll-area/src/ScrollArea.tsx#L183

Running into an issue where I can't stylize that wrapper div so that'll be same size as its parent.

![image](https://user-images.githubusercontent.com/15721634/174700315-bd5ebb7e-2f79-45a1-a462-1f419ae63df0.png)

![image](https://user-images.githubusercontent.com/15721634/174700455-0e0c8ae1-a612-4fba-ae26-ed037e51c9bf.png)

**Comment by andy-hook** [comment]
@BrianHung This is possible for me by specifying `height`:

https://codesandbox.io/s/scroll-area-fill-parent-5kubxo?file=/App.js

Does that work for your case? I might be misunderstanding.

**Reply by BrianHung** [reply]
In the example, when I set `length: 1` such that the child's height is less than parent's height, it doesn't fill up the space.

![image](https://user-images.githubusercontent.com/15721634/174893822-da909e91-16dd-46d5-8103-6b34a97000ca.png)

My use case is with a contenteditable that should grow to fit the height of the parent (even with no content), and scroll when its height from its content exceeds its parent.

**Comment by jamesarosen** [comment]
I have a solution. Make the ScrollAreaViewport a CSS Container (using `container-type: size` or `container-type: scroll-state size`), then set the height of the child to `100cqh`.

Basing on the [Scroll Area demo](https://www.radix-ui.com/primitives/docs/components/scroll-area) from the Radix docs:

```css
.ScrollAreaViewport {
  container-name: scroll-viewport;
  container-type: scroll-state size;
}

.h-100cqh {
  height: 100cqh;
}
```

```jsx
<ScrollArea.Root className="ScrollAreaRoot">
  <ScrollArea.Viewport className="ScrollAreaViewport">
    <div class="h-100cqh">
      This content will take up 100% of the scroll-area viewport height.
    </div>
  </ScrollArea.Viewport>
</ScrollArea.Root>
```

Note: Tailwind's `@container` class applies `container-type: inline-size`, which is great for responsive layouts, but won't set a queryable height that you can use for this case. If you happen to be using Tailwind, you'll need to define a custom utility to set `container-type: size` or `container-type: scroll-state size`.
